### PR TITLE
SY: LabVIEW Client MVP - Part 2 (Error Handling + Channel Create)

### PR DESCRIPTION
**Pull request by sy-nico**
# Issue Pull Request

## Description

Second milestone of the LabVIEW client, covering structured error handling across the C-ABI and channel create/retrieve from LabVIEW.

C-ABI (synnax_clib):

- Returns errors to LabVIEW through a fixed 644-byte SynnaxError struct (i32 code, char[128] type, char[512] message), passed as an Array Data Pointer on every error-returning function. A static_assert and a layout test pin the 644-byte layout the LabVIEW side depends on.
- Adds synnax_channel_create with batch support (1..N channels) via parallel arrays, alongside synnax_channel_retrieve_keys.
- Extracts the newline-splitting marshaling into a shared helper used by retrieve and create.
- Fixes two malformed codec error messages (a stray literal in the data-type mismatch message and a missing space in the extra-key message).

LabVIEW client:

- Adds write I/O terminals for all supported data types (f64, f32, i8/i16/i32/i64, u8/u16/u32/u64, string, timestamp).
- Adds polymorphic create VIs across virtual, persisted, and index kinds and single/batch cardinality, all backed by the single batch create CLFN.
- Implements retrieve-if-name-exists client-side (retrieve, create only the missing subset, merge keys back in input order) to match the TypeScript and Python clients.
- Adds a reusable error builder VI that parses the SynnaxError buffer and folds type and message into the LabVIEW error cluster with the call chain intact.
- Adds array-size validation for per-channel index inputs, plus examples and gitignore updates.

## Basic Readiness

- [x] I have performed a self-review of my code.
- [x] I have added relevant, automated tests to cover the changes.
****

<h3>Greptile Summary</h3>

This PR adds the next LabVIEW client milestone for errors and channel creation. The main changes are:

- Fixed-size C-ABI error buffers for LabVIEW calls.
- Batch channel create and retrieve support through the C wrapper.
- LabVIEW create, retrieve, write, and error-building VIs.
- Validation and tests for channel create edge cases.
- Small C++ codec error-message fixes.

<h3>Confidence Score: 5/5</h3>

This looks safe to merge.

- No blocking issues found in the changed code.
- The latest create changes reject empty batches before the lower create call.
- The create wrapper now checks for missing returned keys before reporting success.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| client/clib/channel/channel.cpp | Adds batch channel creation with validation for empty batches, parsed input counts, and missing created keys. |
| client/clib/internal.h | Adds a shared newline-splitting helper used by channel retrieve and create. |
| client/clib/synnax_labview.h | Updates the LabVIEW-facing import header for fixed error buffers and channel creation. |
| client/clib/types.h | Pins the `SynnaxError` byte layout used by LabVIEW. |
| client/clib/channel/channel_test.cpp | Adds tests for LabVIEW error-buffer handling and channel create validation paths. |

<sub>Reviews (3): Last reviewed commit: ["\[clib\] guard empty and short-response ch..."](https://github.com/synnaxlabs/synnax/commit/5b5c2650ccf701044a8ccd0a8c114429bc6fa2b7) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=40581124)</sub>

<details><summary><h4>Context used (3)</h4></summary>

- Rule used - When making changes to functionality, add correspo... ([source](https://app.greptile.com/synnax/-/custom-context?memory=8b39d83e-3d2b-4dd4-9e14-d0644d6bfe58))

**Learned From**
[synnaxlabs/synnax#1550](https://github.com/synnaxlabs/synnax/pull/1550)
- Rule used - Place unit tests in the appropriate test files tha... ([source](https://app.greptile.com/synnax/-/custom-context?memory=f96c85c8-3531-4c44-9c22-1c7a35161001))

**Learned From**
[synnaxlabs/synnax#1550](https://github.com/synnaxlabs/synnax/pull/1550)
- Context used - CLAUDE.md ([source](https://app.greptile.com/synnax/github/synnaxlabs/synnax/-/custom-context?memory=1f93b7ae-395e-4379-bb71-2f5a9a0f0287))
</details>

### CODE REVIEW

File: client/clib/channel/channel.cpp [hunk 1]
```diff
@@ -114,3 +105,61 @@ int32_t synnax_channel_retrieve_keys(
...
+) {
+    clear_err(err);
+    if (client == nullptr || out_keys == nullptr) {
+        set_err(err, CODE_INTERNAL, "sy.validation", "null client or out_keys");
+        return CODE_INTERNAL;
+    }
+    try {
+        const std::vector<std::string> name_list = split_newlines(names);
+        const std::vector<std::string> dtype_list = split_newlines(data_types);
+        if (name_list.size() != count || dtype_list.size() != count) {
```
[the remark this answers is not in the document]
Comment (sy-nico) [reply client/clib/channel/channel.cpp:137]: Create now rejects count == 0 with a `sy.validation` error before reaching the create call.

File: client/clib/channel/channel.cpp [hunk 2]
```diff
@@ -114,3 +105,61 @@ int32_t synnax_channel_retrieve_keys(
...
+            ch.is_virtual = is_virtual != nullptr && is_virtual[i] != 0;
+            chans.push_back(std::move(ch));
+        }
+        const auto c_err = client->client.channels.create(chans);
+        if (!c_err.ok()) {
+            set_err(err, CODE_ERROR, c_err.type, c_err.data);
+            return CODE_ERROR;
+        }
+        for (size_t i = 0; i < count; i++)
+            out_keys[i] = chans[i].key;
```
[the remark this answers is not in the document]
Comment (sy-nico) [reply client/clib/channel/channel.cpp:174]: Clib now returns a `sy.unexpected` error if any key is still 0 after a successful create, so a short response no longer reports `CODE_OK`.

**Comment by sy-nico** [comment]
@greptile, rerun. Both comments addressed. Please update PR description with updated review.

File: client/clib/channel/channel.cpp [hunk 3]
```diff
@@ -114,3 +105,80 @@ int32_t synnax_channel_retrieve_keys(
...
+) {
+    clear_err(err);
+    if (client == nullptr || out_keys == nullptr) {
+        set_err(err, CODE_INTERNAL, "sy.validation", "null client or out_keys");
+        return CODE_INTERNAL;
+    }
+    if (count == 0) {
+        set_err(
+            err,
+            CODE_INTERNAL,
```
[the remark this answers is not in the document]
Comment (sy-nico) [reply client/clib/channel/channel.cpp:128]: Already handled: count == 0 is rejected with a sy.validation error before the create call (lines 125-128). See reply to other comments.

File: client/clib/channel/channel.cpp [hunk 4]
```diff
@@ -114,3 +105,80 @@ int32_t synnax_channel_retrieve_keys(
...
+        }
+        std::vector<synnax::channel::Channel> chans;
+        chans.reserve(count);
+        for (size_t i = 0; i < count; i++) {
+            synnax::channel::Channel ch;
+            ch.name = name_list[i];
+            ch.data_type = x::telem::DataType{dtype_list[i]};
+            ch.is_index = is_index != nullptr && is_index[i] != 0;
+            ch.index = index != nullptr ? index[i] : 0;
+            ch.is_virtual = is_virtual != nullptr && is_virtual[i] != 0;
```
[the remark this answers is not in the document]
Comment (sy-nico) [reply client/clib/channel/channel.cpp:156]: Already handled: the out_keys loop returns a sy.unexpected error if any chans[i].key is still 0 after create. See reply to other comments.

**Comment by sy-nico** [comment]
@greptile, rerun.
